cypress-tests/cypress/support/commands.js

```diff
@@ -241,6 +242,80 @@ Cypress.Commands.add("confirmCallTest", (confirmBody, details, confirm, globalSt
   });
 });
 
+Cypress.Commands.add("confirmBankTransferCallTest",  (confirmBody, confirm, globalState) => {
+  const paymentIntentID = globalState.get("paymentID");
+  confirmBody.client_secret = globalState.get("clientSecret");
+  confirmBody.confirm = confirm;
+
+  cy.request({
+    method: "POST",
+    url: `${globalState.get("baseUrl")}/payments/${paymentIntentID}/confirm`,
+    headers: {
+      "Content-Type": "application/json",
+      "api-key": globalState.get("publishableKey"),
+    },
+    body: confirmBody,
+  }).then((response) => {
+    logRequestId(response.headers["x-request-id"]);
+    expect(response.headers["content-type"]).to.include("application/json");
+    globalState.set("paymentID", paymentIntentID);
+    if (response.body.capture_method === "automatic") {
+      if (response.body.authentication_type === "three_ds" || response.body.authentication_type === "no_three_ds") {
+        console.log(response.body);
+        if (response.body.payment_method_type === "pix") {
+        expect(response.body)
+        .to.have.property("next_action")
+        .to.have.property("qr_code_url");
+        globalState.set(
+          "nextActionUrl", // This is intentionally kept as nextActionUrl to avoid issues during handleRedirection call,
+          response.body.next_action.qr_code_url
+        );
+        } else {
+          expect(response.body)
+            .to.have.property("next_action")
+            .to.have.property("redirect_to_url");
+          globalState.set(
+          "nextActionUrl",
+          response.body.next_action.redirect_to_url
+        );
+      }
+    } else {
+      // Handle other authentication types as needed
+      throw new Error(
+        `Unsupported authentication type: ${authentication_type}`
+      );
+    }
+  } else if (response.body.capture_method === "manual") {
+    if (response.body.authentication_type === "three_ds" || response.body.authentication_type === "no_three_ds") {
+        if (response.body.payment_method_type === "pix") {
+        expect(response.body)
+          .to.have.property("next_action")
+          .to.have.property("qr_code_url");
+          globalState.set(
+            "nextActionUrl", // This is intentionally kept as nextActionUrl to avoid issues during handleRedirection call
+            response.body.next_action.qr_code_url
+          );
+        } else {
+          expect(response.body)
+            .to.have.property("next_action")
+            .to.have.property("redirect_to_url");
+          globalState.set(
+          "nextActionUrl",
+          response.body.next_action.redirect_to_url
+        );
+      }
+      } else {
+        // Handle other authentication types as needed
+        throw new Error(
+          `Unsupported authentication type: ${authentication_type}`
+        );
+      }
+    } else {
+      throw new Error(`Unsupported capture method: ${capture_method}`);
```

Comment:
These could be optimized to re-use same code, but that [requires some refactoring to be done](https://github.com/juspay/hyperswitch/pull/4583/files). Since my attempts to refactor this was earlier rejected (#4583 & #4477), I have decided **not** to optimize this.

Reply:
Still are these inline with the changes that @Sakilmostak recently made to the framework?